### PR TITLE
Refactor to reuse code from Make module

### PR DESCRIPTION
**Pull request by paf31**
Could you please give this a go and see how it fails? It will fail because of the `M.empty` argument to `buildMakeActions`, but I think it's pretty close.

Thanks!

### CODE REVIEW

File: src/Language/PureScript/Ide/Rebuild.hs [hunk 1]
```diff
@@ -52,28 +43,15 @@ rebuildFile path outpath = do
 
   outputDirectory <- confOutputPath . envConfiguration <$> ask
 
-  let initEnv = foldl' (flip P.applyExternsFileToEnvironment) P.initEnvironment externs
+  let ma = P.buildMakeActions outputDirectory
+                              (M.singleton (P.getModuleName m) (Left P.RebuildAlways))
+                              M.empty {- TODO: add foreign module here, if it exists -}
```
Comment (paf31) [inline src/Language/PureScript/Ide/Rebuild.hs:48]: This is the bit which needs to be fixed.

**Comment by kritzcreek** [comment]
Looks awesome! How would we go about eventually salvaging the Environment we created? I know that's an optimization on top but I'm really curious how fast that would make the rebuild :)

Also one of my tests fails. It's a test where I load a module and not its dependency. It looks like this rebuild still succeeds and I'm not sure how?

**Comment by paf31** [comment]
I think it's just a case of caching the `Environment` in your state object. I can't see why that wouldn't work.

**Comment by kritzcreek** [comment]
Allright great, Thanks! I put error reporting back in in https://github.com/kRITZCREEK/purescript/commit/3aa0fa01e5bd9ade25c655d9e12ba9d7336bf6ff.

About the Environment thing: I meant we'd need to make a version of `runMake` that also returns the produced Environment, but that's easy to add when we need it.

**Comment by paf31** [comment]
I don't think you need to keep the environment from `make` though.

Thanks for the merge! What's left now?

**Comment by kritzcreek** [comment]
hmm maybe I was a bit too hasty... :D I get a bunch of unknown Type errors for Prim Types. Do we need to somehow add those to the environment before rebuilding?

**Comment by paf31** [comment]
Hmm, not sure, I'll have a look in a bit.

**Pull request merged**